### PR TITLE
Add RRPN models for PaddleCV

### PR DESCRIPTION
**Pull request by cjt222**
add rrpn model for detecting orientation target such as text and so on

### CODE REVIEW

**Comment by jiangjiajun** [comment]
PaddleCV/rrpn/models/ext_op/src/rrpn_lib.so 这个文件是编译生成的吧，怎么也提交了上来

File: PaddleCV/rrpn/README.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+│   ├── img_111.jpg
+│   ├── img_112.jpg
+|   ...
+├── ch4_test_localization_transcription_gt
+│   ├── img_111.jpg
+│   ├── img_112.jpg
+|   ...
+```
+### 自定义数据
+原始的RRPN只提供了二分类，若要使用自己数据进行训练多分类，需在utility.py中将dataset改为icdar2017，然后将class_num改为需求类别数。
```
Comment (jerrywgz) [inline PaddleCV/rrpn/README.md]: 注明下class_num是否包含背景类

File: PaddleCV/rrpn/README.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+    - 通过设置`--pretrained_model=${path_to_trained_model}`指定训练好的模型，注意不是初始化的模型。
+    - 通过设置`export CUDA\_VISIBLE\_DEVICES=0`指定单卡GPU评估。
+
+
+下表为模型评估结果：
+
+RRPN
+
+| 模型                   | 批量大小   | 迭代次数   | F1  |
```
Comment (jerrywgz) [inline PaddleCV/rrpn/README.md:141]: 是否需要开源模型？

File: PaddleCV/rrpn/checkpoint.py [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+import shutil
+import time
+import numpy as np
+import re
+import paddle.fluid as fluid
+import logging
+logger = logging.getLogger(__name__)
+
+
+def load_params(exe, prog, path, ignore_params=[]):
```
Comment (jerrywgz) [inline PaddleCV/rrpn/checkpoint.py]: 这个函数是来自检测库吧，ignore_params这里不需要，或者补充该参数的使用文档

File: PaddleCV/rrpn/config.py [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+_C.resolution = 14
+
+#
+# SOLVER options
+#
+
+# derived learning rate the to get the final learning rate.
+_C.learning_rate = 0.01
+
+# maximum number of iterations, 1x: 180000, 2x:360000
```
Comment (jerrywgz) [inline PaddleCV/rrpn/config.py]: 注释部分需要修改1x 2x对应140000 280000？下同

File: PaddleCV/rrpn/data_utils.py [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+    if mode == 'train' or mode == 'val':
+        with open(roidb['image'], 'rb') as f:
+            data = f.read()
+        data = np.frombuffer(data, dtype='uint8')
+        img = cv2.imdecode(data, 1)
+        img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
+        gt_boxes = roidb['boxes']
+        gt_label = roidb['gt_classes']
+        # resize
+        if mode == 'train':
```
Comment (jerrywgz) [inline PaddleCV/rrpn/data_utils.py:68]: 为什么训练测试使用不同resize大小？

File: PaddleCV/rrpn/learning_rate.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import paddle.fluid as fluid
+import paddle.fluid.layers.learning_rate_scheduler as lr_scheduler
+from paddle.fluid.layers import control_flow
+
+
+def exponential_with_warmup_decay(learning_rate, boundaries, values,
```
Comment (jerrywgz) [inline PaddleCV/rrpn/learning_rate.py]: 现在paddle有warmup的学习率https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/fluid/layers/learning_rate_scheduler.py#L460

File: PaddleCV/rrpn/config.py (hunk 4 above)
Comment (cjt222) [reply PaddleCV/rrpn/config.py]: Done，删除了多余注释

File: PaddleCV/rrpn/README.md (hunk 1 above)
Comment (cjt222) [reply PaddleCV/rrpn/README.md]: > PaddleCV/rrpn/models/ext_op/src/rrpn_lib.so 这个文件是编译生成的吧，怎么也提交了上来

Done

File: PaddleCV/rrpn/checkpoint.py (hunk 3 above)
Comment (cjt222) [reply PaddleCV/rrpn/checkpoint.py]: Done

File: PaddleCV/rrpn/learning_rate.py (hunk 6 above)
Comment (cjt222) [reply PaddleCV/rrpn/learning_rate.py]: Done

File: PaddleCV/rrpn/README.md (hunk 2 above)
Comment (cjt222) [reply PaddleCV/rrpn/README.md:141]: Done，已添加开源模型

**Review by jerrywgz — COMMENTED** [review]
LGTM

**Review by qingqing01 — APPROVED** [review]
